src/handler/index.ts

```diff
@@ -35,45 +35,43 @@ export const handler = async (event: DynamoDBStreamEvent, context: Context): Pro
   const emailMessages: EmailMessageRequest[] = [];
   logger.info(`Received ${records.length} records from DynamoDB.`);
   records.forEach((record) => {
-    // Discard INSERT and REMOVE events
-    if (record.eventName !== 'MODIFY') {
-      logger.info(`Discarding ${record.eventName} event.`);
+    logger.debug(`Processing record: ${JSON.stringify(record)}`);
+
+    let message: AvailabilityChangeData;
+    try {
+      const body: Record<string, string> = <Record<string, string>> JSON.parse(record.body);
+      message = JSON.parse(body.Message) as AvailabilityChangeData;
+    } catch (parseError) {
+      logger.error(`Error while parsing message body for messageId: ${record.messageId}`);
+      logger.error(parseError);
       return;
     }
 
-    logger.debug(`Processing record: ${JSON.stringify(record)}`);
-
     // Extract data from event
     let availabilityData: AvailabilityChangeData;
     try {
-      availabilityData = extractAvailabilityData(record);
+      availabilityData = validateAvailabilityData(message);
```

Comment:
As per discussion - shall we change this ? and consolidate the error handling

Reply:
Consolidated the try catch and handled the different errors by checking the type of error thrown